recipients_app/lib/core/cubits/survey/survey_cubit.dart

```diff
@@ -1,74 +1,134 @@
 import "package:app/data/models/models.dart";
+import "package:app/data/models/survey/mapped_survey.dart";
+import "package:app/data/models/survey/survey_card_status.dart";
 import "package:app/data/repositories/crash_reporting_repository.dart";
+import "package:app/data/repositories/survey_repository.dart";
 import "package:equatable/equatable.dart";
-import "package:firebase_auth/firebase_auth.dart";
 import "package:flutter_bloc/flutter_bloc.dart";
 
 part "survey_state.dart";
 
+const _kNewSurveyDay = -10;
+const _kNormalSurveyStartDay = 0;
+const _kNormalSurveyEndDay = 10;
+const _kOverdueEndDay = 15;
+const _kOverduePeriodDays = 6;
+const _kEndOfDisplaySurveyDay = 20;
+
 class SurveyCubit extends Cubit<SurveyState> {
   final Recipient recipient;
-  final User currentFirebaseUser;
+  final SurveyRepository surveyRepository;
   final CrashReportingRepository crashReportingRepository;
 
   SurveyCubit({
     required this.recipient,
-    required this.currentFirebaseUser,
+    required this.surveyRepository,
     required this.crashReportingRepository,
   }) : super(const SurveyState());
 
-  /// TODO: add logic for getting the survey
-  void getSurveyUrl() {
-    final creationDate = currentFirebaseUser.metadata.creationTime;
-    final nextSurveyDate = recipient.nextSurvey!.toDate();
+  void getSurveys() async {
+    final surveys =
+        await surveyRepository.fetchSurveys(recipientId: recipient.userId);
+    final mappedSurveys = surveys
```

Comment:
I think it makes sense to wrap this in try catch and then in the catch bloc also log the error to the crashreportingrepo?

Reply:
fixed [db61371](https://github.com/socialincome-san/public/pull/413/commits/db613717cad931f1ff6cdb0df3dbdfe387ec6bb0)